haddock-library/src/Documentation/Haddock/Parser/Util.hs

```diff
@@ -13,60 +13,73 @@
 module Documentation.Haddock.Parser.Util (
   unsnoc
 , strip
+, inClass
+, notInClass
 , takeUntil
 , removeEscapes
 , makeLabeled
 , takeHorizontalSpace
 , skipHorizontalSpace
 ) where
 
+import qualified Data.Text as T
+import           Data.Text (Text)
+
 import           Control.Applicative
 import           Control.Monad (mfilter)
-import           Documentation.Haddock.Parser.Monad hiding (isHorizontalSpace)
-import           Data.ByteString.Char8 (ByteString)
-import qualified Data.ByteString.Char8 as BS
+import           Documentation.Haddock.Parser.Monad
 import           Prelude hiding (takeWhile)
 
-#if MIN_VERSION_bytestring(0,10,2)
-import           Data.ByteString.Char8 (unsnoc)
-#else
-unsnoc :: ByteString -> Maybe (ByteString, Char)
+import           Data.Maybe (isJust)
+import           Data.Char (isSpace)
+
+-- | Like 'T.uncons', but for the last character instead of the first.
+unsnoc :: Text -> Maybe (Text, Char)
 unsnoc bs
-  | BS.null bs = Nothing
-  | otherwise = Just (BS.init bs, BS.last bs)
-#endif
+  | T.null bs = Nothing
+  | otherwise = Just (T.init bs, T.last bs)
 
 -- | Remove all leading and trailing whitespace
-strip :: String -> String
-strip = (\f -> f . f) $ dropWhile isSpace . reverse
+strip :: Text -> Text  
+strip = T.strip 
```

Comment:
As in other module, we don't really care about this function, we can just use T.strip at call site instead.